doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +122,80 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
 Configuring your application
 ****************************
 
-If your application uses a provided library or targets a specific board, you might want to change the default configuration of the library or board.
+You might want to change the default options of the application.
 There are different ways of doing this, but not all will store your configuration permanently.
 
-The default configuration for a library is specified in its :file:`Kconfig` file.
-Similarly, the default configuration for a board is specified in its :file:`*_defconfig` file (and its :file:`Kconfig.defconfig` file, see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
-The configuration for your application, which might override some default options of the libraries or the board, is specified in a :file:`prj.conf` file in the application directory.
+Zephyr's configuration system is distributed, and so is the |NCS| one.
+This means that there are multiple sources of configuration.
+In the |NCS|, the configuration system consists of the following elements:
+
+* Kconfig configuration files that focus on software-related options.
+* Devicetree specification (DTS) configuration files that focus on hardware-related options.
```

Comment:
I wouldn't call devicetree files `configuration files`.
The main purpose of devicetree files is to describe the hardware.

But I believe @mbolivar-nordic would be better than me on how to phrase this in a precise and correct manner.

Reply:
But DTS files are configuration files. You actually specify a lot of things using DTS files that are reflected in either what FW does (e.g. what component is choses to use), which partition it uses or even what speeds/channels/pins HW should use for the sake of the application. It's something that integrator will change freely depending on needs.

Reply:
The parts of device tree that users work with the most is most definitely configuration.

I wish device tree only did HW description.

Reply:
> but I would appreciate input from @mbolivar-nordic on this.

DT is definitely configuration too. This is in the first paragraph of the "introduction to devicetree" page in the zephyr docs:

"Zephyr uses devicetree to describe the hardware available on its Supported Boards, **as well as that hardware’s initial configuration**."

https://docs.zephyrproject.org/latest/guides/dts/intro.html

Reply:
(Somewhat related, I actually plan on moving the devicetree documentation pages in the zephyr docs under the new "build and configuration systems" heading)

Reply:
No changes made (well, I added the link, but it was not the point of this specific conversation).